### DISCUSSION TITLE
Packagist.org API for downloading the complete dump of PHP package metadata in one go

### DISCUSSION THREAD

**Discussion by nareshdevasani** [composer/composer#11327] [Q&A / Support] [answered]
Packagist.org provides APIs to query package metadata https://packagist.org/apidoc
It is good but I am looking for an API which can give a complete dump of package metadata.

Below API gives all packages (only names)
GET https://packagist.org/packages/list.json 

and below API gives metadata for the given package.
GET https://repo.packagist.org/p2/[vendor]/[package].json

What I am looking for is to get the metadata for all the packages as a single archive file.

**Comment by stof** [comment, marked as answer]
There is no endpoint giving all metadata about all packages at once (which would be huge and would take a long time to dump)
